File: frontend/src/types/portfolio.ts
export type PortfolioAsset = {
  id: number;
  symbol: string;
  asset_type: string;
  quantity: number;
  buy_price: number;
};

export type PortfolioCreatePayload = {
  symbol: string;
  asset_type: string;
  quantity: number;
  buy_price: number;
};

export type PortfolioSummary = {
  user: string;
  asset_count: number;
  total_cost: number;
  currency: string;
};

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel


class PortfolioAssetCreate(BaseModel):
    symbol: str
    asset_type: str = "crypto"
    quantity: float
    buy_price: float


class PortfolioAssetResponse(BaseModel):
    id: int
    symbol: str
    asset_type: str
    quantity: float
    buy_price: float

    class Config:
        from_attributes = True


class PortfolioSummary(BaseModel):
    total_cost: float
    total_value: float
    profit_loss: float
    profit_loss_percent: float